### PR TITLE
Add opt-in property prefix support to Java and C#

### PR DESCRIPTION
**Pull request by externl**
See #2855

### CODE REVIEW

File: csharp/src/Ice/Properties.cs [hunk 1]
```diff
@@ -37,6 +37,11 @@ public Properties()
     {
     }
 
+    public Properties(List<string> optInPrefixes)
+    {
+        _optInPrefixes = optInPrefixes;
+    }
```
Comment (pepone) [inline csharp/src/Ice/Properties.cs:43]:
```suggestion
    public Properties(List<string> optInPrefixes) => _optInPrefixes = optInPrefixes;
```

File: csharp/src/Ice/Properties.cs [hunk 2]
```diff
@@ -55,6 +60,11 @@ public Properties(ref string[] args, Properties? defaults = null)
             {
                 _properties[entry.Key] = entry.Value.Clone();
             }
+
+            foreach (string prefix in defaults._optInPrefixes)
+            {
+                _optInPrefixes.Add(prefix);
+            }
```
Comment (pepone) [inline csharp/src/Ice/Properties.cs:67]: I think we can use AddRange here
```suggestion
            _optInPrefixes.AddRange(defaults._optInPrefixes);
```

File: csharp/src/Ice/Properties.cs [hunk 3]
```diff
@@ -324,6 +334,12 @@ public void setProperty(string key, string value)
         // Check if the property is in an Ice property prefix. If so, check that it's a valid property.
         if (findIcePropertyArray(key) is PropertyArray propertyArray)
         {
+            if (propertyArray.isOptIn && !_optInPrefixes.Contains(propertyArray.name))
+            {
+                throw new PropertyException($"unable to set '{key}': property prefix '{propertyArray.name}' is " +
+                "opt-in and must be explicitly enabled");
```
Comment (pepone) [inline csharp/src/Ice/Properties.cs:340]: In C# we prefer the message in a single line even if exceed the max line length
```suggestion
                throw new PropertyException(
                    $"unable to set '{key}': property prefix '{propertyArray.name}' is opt-in and must be explicitly enabled");
```

File: csharp/src/IceBoxNet/ServiceManagerI.cs [hunk 4]
```diff
@@ -896,21 +896,19 @@ public StartServiceInfo(string service, string value, string[] serverArgs)
 
     private Ice.Properties createServiceProperties(string service)
     {
-        Ice.Properties properties;
+        Ice.Properties properties = new Ice.Properties();
```
Comment (pepone) [inline csharp/src/IceBoxNet/ServiceManagerI.cs:899]:
```suggestion
        var properties = new Ice.Properties();
```

File: csharp/src/IceBoxNet/ServiceManagerI.cs [hunk 5]
```diff
@@ -896,21 +896,19 @@ public StartServiceInfo(string service, string value, string[] serverArgs)
...
         if (communicatorProperties.getIcePropertyAsInt("IceBox.InheritProperties") > 0)
         {
-            properties = communicatorProperties.Clone();
-            // Inherit all except Ice.Admin.xxx properties
-            foreach (string p in properties.getPropertiesForPrefix("Ice.Admin.").Keys)
+            // Inherit all except IceBox. and Ice.Admin. properties
+            foreach (var property in communicatorProperties.getPropertiesForPrefix(""))
             {
-                properties.setProperty(p, "");
+                if (!property.Key.StartsWith("IceBox.", StringComparison.Ordinal) && !property.Key.StartsWith("Ice.Admin.", StringComparison.Ordinal))
```
Comment (pepone) [inline csharp/src/IceBoxNet/ServiceManagerI.cs:906]: line too long

File: java/src/IceGridGUI/src/main/java/com/zeroc/IceGridGUI/Coordinator.java
Comment (pepone) [inline java/src/IceGridGUI/src/main/java/com/zeroc/IceGridGUI/Coordinator.java:1]: Why are many unrelated formatting changes in this file

**Review by pepone — APPROVED** [review]

File: java/src/IceGridGUI/src/main/java/com/zeroc/IceGridGUI/Coordinator.java
Comment (externl) [reply java/src/IceGridGUI/src/main/java/com/zeroc/IceGridGUI/Coordinator.java:1]: Hmm...not sure. I only modified one line then saved and reformatted. I'll see if I can figure out what happened.

Comment (externl) [reply java/src/IceGridGUI/src/main/java/com/zeroc/IceGridGUI/Coordinator.java:1]: Should be fixed.
